### PR TITLE
Workaround to improve formatting of html code in biodiversity zones

### PR DESCRIPTION
**Pull request by momomaniac**
according to finding in email, html entities are decoded correctly

### CODE REVIEW

File: src/components/map/BiodivInformation.vue [hunk 1]
```diff
@@ -14,15 +14,16 @@
       <span>{{ months.join(", ") }}</span>
     </p>
 
-    <!-- TODO : find a light way to securely show this content -->
-    <p>{{ description }}</p>
+    <!-- content formatting done with "he" html entities decoder -->
+    <p v-for="descPar of descriptionParagraphs">{{ descPar }}</p>
```
Comment (cbeauchesne) [inline src/components/map/BiodivInformation.vue]: `v-for` needs a `key` attribute

```suggestion
    <p v-for="(descPar, i) of descriptionParagraphs" :key="i">{{ descPar }}</p>
```

File: src/components/map/BiodivInformation.vue [hunk 2]
```diff
@@ -14,15 +14,16 @@
...
     <a :href="data.infoUrl" target="_blank" rel="noopener" v-translate>More info</a> |
     <a :href="data.kmlUrl" target="_blank" rel="noopener">KML</a>
   </modal-window>
 </template>
 
 <script>
 
+  import he from 'he';
```
Comment (cbeauchesne) [inline src/components/map/BiodivInformation.vue]: I'm surprised to see this lib, we removed from our stack 6 month ago as it's very heavy. It works on your end???

For removing HTML entities , we have a function in utils : `utils.decodeHtmlEntities()` it also removes html markup, so you can skip discard all remaining html tags.

**Comment by cbeauchesne** [comment]
related to #904 , thanks @momomaniac 

Small hint : run `npm run lint` before committing, it'll clean a lot of CI issues :)

**Comment by momomaniac** [comment]
thanks a lot for your remarks.
The 'he' library worked for me, but I'm glad, you had some project history. No point putting a heavy library back into the code...

Working for me now, with quite acceptable rendering results.

File: src/components/map/BiodivInformation.vue [hunk 3]
```diff
@@ -14,15 +14,16 @@
       <span>{{ months.join(", ") }}</span>
     </p>
 
-    <!-- TODO : find a light way to securely show this content -->
-    <p>{{ description }}</p>
+    <!-- content formatting done with "he" html entities decoder -->
```
Comment (brunobesson) [inline src/components/map/BiodivInformation.vue]: this comment is then obsolete

File: src/components/map/BiodivInformation.vue [hunk 4]
```diff
@@ -32,12 +33,15 @@
...
 
-        // very simple html > text interpreter : only html entities
-        result = result.replace(/&eacute;/g, 'é');
-        result = result.replace(/&egrave;/g, 'è');
+        // keep line breaks for better readability
+        result = result.replace(/<br ?\/?>/g, '%%MYNL%%');
+        // decode html entities
+        result = utils.decodeHtmlEntities(result);
+        // restore initial line breaks
+        result = result.split('%%MYNL%%');
```
Comment (brunobesson) [inline src/components/map/BiodivInformation.vue]: you're lucky we are not doing any type checking... but ok, it works.
for readability, I would prefer to use another variable for the returned array

**Comment by momomaniac** [comment]
first version was rather a quick hack, now it's way cleaner

File: src/components/map/BiodivInformation.vue [hunk 5]
```diff
@@ -32,13 +33,15 @@
...
-
-        // very simple html > text interpreter : only html entities
-        result = result.replace(/&eacute;/g, 'é');
-        result = result.replace(/&egrave;/g, 'è');
-        return result;
+      descriptionParagraphs() {
+        // keep line breaks for better readability
+        const paragraphList = (this.data.description || '').split(/<br ?\/?>/);
+        // decode html entities
+        let i = 0;
```
Comment (brunobesson) [inline src/components/map/BiodivInformation.vue]: a little bit of functional programming would simplify the whole:

```js
return (this.data.description || '')
  .split(/<br ?\/?>/)
  .map(p => utils.decodeHtmlEntities(p);
```

(although I don't remember which es version we support on c2c_ui)

**Comment by cbeauchesne** [comment]
Ok for me, can I merge @momomaniac  ?

**Comment by momomaniac** [comment]
ok for me ;-)

**Pull request merged**